### ISSUE TITLE
Terraform fails when trying to assign a role to an Azure Web App

### ISSUE THREAD

**Issue by giovannifl** [terraform-providers/terraform-provider-azurerm#12194]
While running Terraform to assign the "AcrPull" role ion Azure Container Registry to an Azure Web App, execution failes with the  following error:

```
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to azurerm_role_assignment.acr_Pull, provider "provider[\"registry.terraform.io/hashicorp/azurerm\"]" produced an unexpected new value: Root resource was present, but now absent.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker. 
```

Below there is the affected code:

```
data "azurerm_container_registry" "acr" {
  name                = var.acrName
  resource_group_name = var.acrRGName
}

resource "azurerm_role_assignment" "acr_Pull" {
  scope                = data.azurerm_container_registry.acr.id
  role_definition_name = "AcrPull"
  principal_id         = azurerm_app_service.wap.identity[0].principal_id
}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.0.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.62.0

### Affected Resource(s)

* `azurerm_container_registry`
* `azurerm_role_assignment`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

data "azurerm_container_registry" "acr" {
  name                = var.acrName
  resource_group_name = var.acrRGName
}

resource "azurerm_role_assignment" "acr_Pull" {
  scope                = data.azurerm_container_registry.acr.id
  role_definition_name = "AcrPull"
  principal_id         = azurerm_app_service.wap.identity[0].principal_id
}

### Debug Output
azurerm_role_assignment.acr_Pull: Creating...
azurerm_role_assignment.acr_Pull: Still creating... [10s elapsed]
azurerm_role_assignment.acr_Pull: Still creating... [20s elapsed]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to azurerm_role_assignment.acr_Pull, provider "provider[\"registry.terraform.io/hashicorp/azurerm\"]" produced an unexpected new value: Root resource was present, but now absent.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by OmerLaza** [comment]
Hey @giovannifl
I just had this problem yesterday on #12195 and @tombuildsstuff helped me. It seems you'll need to upgrade your provider version to the newest v2.63.0 (or just v2.62.1) or downgrade to older version since there was a regression in v2.62.0.

Hope this will help 😄

**Comment by katbyte** [comment]
thanks @OmerLaza - as this has been fixed in 2.63 i'm going to close this
